### PR TITLE
Refactor and implement new status for ticket

### PR DESCRIPTION
**Pull request by swomfire**
close https://github.com/miaconsults/mia/issues/100
- Update status enum for ticket
- Create timer for idle, disconnect.
- Refactor status flow

### CODE REVIEW

File: server/modules/agent/agent.controller.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
 import httpStatus from 'http-status';
+import _ from 'lodash';
```
Comment (longhpzigvy) [inline server/modules/agent/agent.controller.js]: `import assign from 'lodash/assign';`

File: server/modules/reply/reply.controller.js [hunk 2]
```diff
@@ -59,6 +65,12 @@ class ReplyController extends BaseController {
       } else {
         setTimeout(() => this.getResponseFromMia(userReply), 0);
       }
+      const { ticketId } = await ConversationService.getOneByQuery({ _id: conversationId });
+      TicketService.update(ticketId, { status: TICKET_STATUS.PROCESSING });
+      IdleQueue.destroyTimer(ticketId);
```
Comment (longhpzigvy) [inline server/modules/reply/reply.controller.js]: Can we write a function that called `resetTimer`?

Comment (swomfire) [reply server/modules/reply/reply.controller.js]: ok

File: server/socketio/index.js [hunk 3]
```diff
@@ -2,20 +2,33 @@ import createSocketIO from 'socket.io';
...
   constructor(authenticate) {
     this.authenticate = authenticate;
   }
 
   connect(server) {
-    const socketIO = createSocketIO(server, {
+    socketIO = createSocketIO(server, {
       path: '/chat',
     });
+    this.socketIO = socketIO;
```
Comment (longhpzigvy) [inline server/socketio/index.js]: useless old code?

**Pull request merged**